### DISCUSSION TITLE
Can't compile newest commit, and no option for RS485 battery in previous or current build

### DISCUSSION THREAD

**Discussion by daJokur** [skippermeister/OpenDTU-OnBattery#13] [Q&A] [answered]
Writing in English so everybody can participate but am native German speaker, so feel free to answer in German.

I just recently started looking into OpenDTU-OnBattery, I currently have the original OpenDTU running successfully since my setup is fairly simple (Hoymiles HMS-2000-4T, Victron SmartSolar 150/30, Pylontech-compatible battery Felicity Solar LPBA48100-OL). But since I want to read the state of charge of my Pylontech-compatible, I wanted to try this fork. That's where my problems start:
- I can't compile the current version (2024-09-28) for any board. The log shows errors about yarn not being a recognized command. I tried installing additional extensions, installed NodeJS/npm on my machine, and manually tried installing yarn/lint, without success. 
- Compiling the older version (2024-09-07) as well as updating the firmware via the web interface worked so far, but I never am finding the possibility to turn on the battery interface via RS485, only for CAN0. Do I have to upload an adjusted pin_mapping.json file? 

Thanks in advance for any help 😊 

Edit: tried using the pin mapping file from [here](https://github.com/skippermeister/OpenDTU-OnBattery/discussions/1), but it made my ESP32-S3 freeze and I had to erase the flash. After changing nearly every pin to -1, it worked.

**Comment by skippermeister** [comment, marked as answer]
@lehmartin 
Warum yarn und Lint nicht funktioniert hatte meine ich mit einem Update der yarn und lint Versionen zu tun. Ich hatte ein ähnliches Problem und habe platformio komplett neu installiert. Hatte ich viel Zeit dran verbracht. Was es genau war kann ich aber nicht mehr sagen.
Die Release Erstellung im Github funktioniert ja und auch bei mir lokal. Dürfte also kein generelles Problem sein.

bzgl. Pinmapping. Wenn Geräte im pinmapping falsch konfiguriert sind und aktiviert werden sollen, kann ein freeze passieren, insbesondere wenn im pinmapping Pins verwendet werden, die beim verwendeten ESP32S3 auf die Pins des PSRAM verweisen. Schick mal bitte dein jetziges pinmapping und ein log der Bootphase.
An welche Pins hast Du welche Geräte angeschlossen?

**Comment by daJokur** [comment]
Danke für die Antwort! Dann probiere ich es bei Gelegenheit mit einer Neuinstallation. 

Dass mein Pinmapping fehlerhaft habe ich auch feststellen müssen, hat geklappt nachdem ich es angepasst habe. Kann dir gerade keine pinmapping file zukommen lassen da ich wieder zum stock OpenDTU (nicht onBattery) zurück bin weil ich die extra Funktionen doch nicht brauche. Dachte mein Speicher sei Pylontech-kompatibel, aber scheinbar nur über CAN, nicht über RS485. Habe die Dokumentation zum RS485-Protokoll von Felicity-Solar angefragt und es selber implementiert. Kann bei Bedarf mehr Infos dazu teilen, aber vermute die wenigsten hier haben Akkus von Felicity 😄

**Reply by PPolo11** [reply]
Hallo lehmartin habe auch Felicity batterie und würde die auch gerne über die RS485 Abfragen,  kannst du mir was zu Verfügung stellen???

**Reply by daJokur** [reply]
Hiya und sorry für die späte Antwort, hatte deine Anfrage nicht vergessen. Hab meinen Code heute mal hochgeladen, kannst ihn gerne anschauen. 
Disclaimer: 
- Ich bin kein Software Engineer, bin Laie 
- Ich habe nur meinen Code hochgeladen, keinerlei Dokumentation. Am besten Diff nutzen um zu sehen wo ich was hinzugefügt hab
- Bei mir funktioniert es, aber keine Garantie dass es bei anderen auch funktionieren wird. 
- Wenn Bedarf da ist, kann ich bei Gelegenheit noch ein paar erklärende Worte dazu geben, fühle mich nur im Moment nicht danach, sorry

Hoffe dennoch, dass mein Code hilft. Here we go: [Mein OpenDTU-Fork (**nicht** OnBattery)](https://github.com/lehmartin/OpenDTU/tree/felicityRS485)

**Reply by PPolo11** [reply]
Danke
